spec/services/processors/blog_post_views_updater_spec.rb

```diff
@@ -0,0 +1,78 @@
+require 'rails_helper'
+
+describe Processors::BlogPostViewsUpdater do
+  describe '.call' do
+    let(:blog_post) { create(:blog_post, published_at: publish_date) }
+    let(:blog_post_views_payload) do
+      create(:blog_post_views_payload, publish_datetime: publish_date).with_indifferent_access
+    end
+    let(:publish_date) { Time.zone.now.beginning_of_month }
+    let(:current_month_views) do
+      blog_post_views_payload['years'][publish_date.year.to_s]['months'][publish_date.month.to_s]
+    end
+    let(:api_service) { instance_double(WordpressService) }
+
+    before do
+      allow_any_instance_of(described_class).to receive(:wordpress_service).and_return(api_service)
+      allow(api_service)
+        .to receive(:blog_post_views)
+        .with(blog_post.blog_id)
+        .and_return(blog_post_views_payload)
+    end
+
+    it 'creates views metrics for the blog post' do
+      described_class.call(blog_post.blog_id)
+
+      created_metric = Metric.find_by(ownable: blog_post)
```

Comment:
I'd add the `name` condition to this query instead of an `expect`

Otherwise in the future when we have more metrics related to Blog posts, this test will fail

Reply:
I think it shouldn't if we keep the DB clean, but nevertheless I'd rather do what you said just in case